File: AutoJump.Core/Game/Device/IDevice.vb
Imports System.Drawing
Imports System.Numerics
''' <summary>
''' 设备接口
''' </summary>
Public Interface IDevice
    ''' <summary>
    ''' 设备是否生效
    ''' </summary>
    ReadOnly Property Availiable As Boolean
    ''' <summary>
    ''' 获取此设备的屏幕大小（以像素为单位）
    ''' </summary>
    ReadOnly Property Size As Size
    ''' <summary>
    ''' 按压
    ''' </summary>
    Sub Press(position As Vector2, millionseconds As Integer)
    ''' <summary>
    ''' 获取设备屏幕图像
    ''' </summary>
    Function GetScreenImage() As Image
End Interface

File: AutoJump.Core/Game/Device/IDevice.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Numerics;
using System.Text;
using System.Threading.Tasks;

namespace AutoJump.Core
{
    /// <summary>
    /// 设备接口
    /// </summary>
    public interface IDevice
    {
        /// <summary>
        /// 设备是否生效
        /// </summary>
        bool Availiable { get; }
        /// <summary>
        /// 获取此设备的屏幕大小（以像素为单位）
        /// </summary>
        Size Size { get; }
        ///<summary>
        ///按压
        ///</summary>
        void Press(Vector2 position, int millionseconds);
        /// <summary>
        /// 返回设备屏幕图像
        /// </summary>
        Image GetScreenImage();
    }
}
